platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/command/ProfileInstallCommand.java

```diff
@@ -0,0 +1,195 @@
+                    .map(Dependency::getName)
+                    .collect(Collectors.toList());
+            startApps(applicationService, profileApps);
+            uninstallInstallerModule(featuresService);
+            printSuccess("Installation Complete");
+        } else {
+            if (optionalExtraProfiles.isPresent()) {
+                Map<String, List<String>> profile = optionalExtraProfiles.get();
+                startApps(applicationService, profile.get(ADVANCED_PROFILE_START_APPS));
+                installFeatures(featuresService, profile.get(ADVANCED_PROFILE_INSTALL_FEATURES));
+                uninstallFeatures(featuresService, profile.get(ADVANCED_PROFILE_UNINSTALL_FEATURES));
+                /* The stop-bundles operation is a workaround currently in place for dealing with
+                    the inter-dependencies of some features, this will likely be removed in the future. */
+                stopBundles(bundleService, profile.get(ADVANCED_PROFILE_STOP_BUNDLES));
+                uninstallInstallerModule(featuresService);
+                printSuccess("Installation Complete");
+            } else {
+                printError(String.format("Profile: %s not found", profileName));
+            }
+        }
+    }
+
+    private void startApps(ApplicationService applicationService, List<String> startupApps)
+            throws ApplicationServiceException {
+        if (startupApps != null) {
+            printSectionHeading("Starting Applications");
+            startupApps.stream().distinct().forEach(application -> {
+                printItemStatusPending("Starting: ", application);
+                executeAsSystem(() -> {
+                    try {
+                        applicationService.startApplication(application);
+                    } catch (ApplicationServiceException e) {
+                        printItemStatusFailure("Start Failed: ", application);
+                        throw new ApplicationServiceException(e);
+                    }
+                    return true;
+                });
+                printItemStatusSuccess("Started: ", application);
+            });
+        }
+    }
+
+    private void uninstallInstallerModule(FeaturesService featuresService) {
+        try {
+            if (!featuresService.isInstalled(featuresService.getFeature("admin-post-install-modules"))) {
+                printSectionHeading("Finalizing Installation");
+                featuresService.installFeature("admin-post-install-modules",
+                        NO_AUTO_REFRESH);
+            }
+            if (featuresService.isInstalled(featuresService.getFeature("admin-modules-installer"))) {
+                featuresService.uninstallFeature("admin-modules-installer",
+                        NO_AUTO_REFRESH);
+            }
+        } catch (Exception e) {
+            printError("An error occurred while trying to perform post-install operations");
```

Comment:
Done, almost all exceptions bubble up to the `doExecute` method now and it prints a warning about the system state if any of them occurs. They are all then re-thrown so that the shell environment can just handle them. Only reason we are now catching them at all is to provide some shell output to the user based on what happened